### PR TITLE
File structure updates

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: architecture/file_structure.rst [hunk 1]
```diff
@@ -1,186 +1,288 @@
...
+    * - libmbedtls/include/
+      - Include files with configuration of mbedtls
 
-/core/include
-*************************
-.. list-table:: Structure of /core/include
-    :header-rows: 1
-    :widths: 1 5
+    * - libmbedtls/mbedtls/
+      - Top directory of the imported mbedtls source tree
```
Comment (jbech-linaro) [inline architecture/file_structure.rst:204]:
```suggestion
      - Top directory of the imported Mbed TLS source tree
```

File: architecture/file_structure.rst [hunk 2]
```diff
@@ -1,186 +1,288 @@
...
 
-    * - Directory
-      - Description
+    * - libmbedtls/mbedtls/include/
+      - Mbedtls include files
```
Comment (jbech-linaro) [inline architecture/file_structure.rst:207]:
```suggestion
      - Mbed TLS include files
```

File: architecture/file_structure.rst [hunk 3]
```diff
@@ -1,186 +1,288 @@
...
 
-    * - /drivers
-      - Include files exposing API for /core/drivers files
+    * - libmbedtls/mbedtls/library/
+      - Mbedtls implementation
```
Comment (jbech-linaro) [inline architecture/file_structure.rst:210]:
```suggestion
      - Mbed TLS implementation
```

File: architecture/file_structure.rst [hunk 4]
```diff
@@ -1,186 +1,288 @@
...
+    * - libunw/include/
+      - Include files for libunwnd
 
-    * - Directory
-      - Description
+    * - libutee/
+      - Libutee which provide the implementation of TEE Internal Core API.
+
+    * - libutee/arch/
+      - Architecure specific implementation
```
Comment (jbech-linaro) [inline architecture/file_structure.rst:222]:
```suggestion
      - Architecture specific implementation
```

File: architecture/file_structure.rst [hunk 5]
```diff
@@ -1,186 +1,288 @@
...
 
-    * - /
-      - Source files for the library
+    * - libutils/isoc/arch/
+      - Architecture specific
 
-    * - /include  
-      - Include files exposing the API of the library
+    * - libutils/isoc/include/
+      - Header files relating to the provided subset of ISOC
```
Comment (jbech-linaro) [inline architecture/file_structure.rst:247]:
```suggestion
      - Header files related to the provided subset of ISOC
```

File: architecture/file_structure.rst [hunk 6]
```diff
@@ -1,186 +1,288 @@
...
+      - Header file of the ABI provided by the trusted key TA
+
+    * - arch
+      - Architecture specific files needed to compile a TA
+
+    * - mk
+      - Makefile includes needed to build TAs and the TA dev kit
+
+    * - avb
+      - TA to support AVB (Android Verified Boot)
```
Comment (jbech-linaro) [inline architecture/file_structure.rst:275]: Worth pointing out that it's AVB 2.0 that we adhere to?

File: architecture/file_structure.rst [hunk 7]
```diff
@@ -1,186 +1,288 @@
...
 
-    * - /arch	  
-      - Architecture specific source files
+    * - pkcs11
+      - TA to support PKCS#11 
 
-    * - /arch/arm 
-      - ARMv7 and Aarch32 specific source files
+    * - pkcs11/src
+      - Source code of the pkcs11 TA
```
Comment (jbech-linaro) [inline architecture/file_structure.rst:284]:
```suggestion
      - Source code for the PKCS#11 TA
```

File: architecture/file_structure.rst [hunk 8]
```diff
@@ -1,186 +1,288 @@
...
 
-    * - /arch/arm/include 
-      - ARMv7 and Aarch32 specific include files
+    * - pkcs11/include
+      - Header file of the ABI provided by the pkcs11 TA
```
Comment (jbech-linaro) [inline architecture/file_structure.rst:287]:
```suggestion
      - Header file for the ABI provided by the PKCS#11 TA
```

File: architecture/file_structure.rst [hunk 9]
```diff
@@ -1,186 +1,288 @@
...
 
-    * - /kta
-      - Include files exposing the KTA API that is mainly used by kernel TAs
+    * - libdl/include/
+      - Include files for libdl
 
-    * - /mm
-      - Include files exposing API for /core/arch/arm/mm files
+    * - libmbedtls/
+      - Mbedtls crypto library
```
Comment (jbech-linaro) [inline architecture/file_structure.rst:194]:
```suggestion
      - Mbed TLS crypto library
```

File: architecture/file_structure.rst [hunk 10]
```diff
@@ -1,186 +1,288 @@
...
-    * - /rom
-      - Old ROM files that should be removed before going public
+    * - libmbedtls/core/
+      - Glue code only compiled with core to connect with the core internal
+        <crypto/crypto.h> API.
 
-    * - /sm
-      - Include files exposing API for Secure Monitor
+    * - libmbedtls/include/
+      - Include files with configuration of mbedtls
```
Comment (jbech-linaro) [inline architecture/file_structure.rst:201]:
```suggestion
      - Include files with configuration of Mbed TLS
```

File: architecture/file_structure.rst (hunk 1 above)
Comment (jbech-linaro) [inline architecture/file_structure.rst:204]:
```suggestion
      - Top directory of the imported Mbed TLS source tree
```

**Review by jbech-linaro — COMMENTED** [review]
Thanks Jens! Mostly minor stuff, with comments addressed.
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: architecture/file_structure.rst (hunk 6 above)
Comment (jenswi-linaro) [reply architecture/file_structure.rst:275]: This might age better without such details.

Comment (jbech-linaro) [reply architecture/file_structure.rst:275]: Sure

**Comment by jenswi-linaro** [comment]
Comments addressed and tag applied. In the process discovered a few more things taken care of in the new commit "[review] architecture: file_structure: update to reflect the current". I'll squash it into "architecture: file_structure: update to reflect the current layout" if that's OK.

**Comment by jbech-linaro** [comment]
> I'll squash it into "architecture: file_structure: update to reflect the current layout" if that's OK.

That's fine, feel free to add my RB.

**Comment by jenswi-linaro** [comment]
Squashed with tag applied.

**Pull request merged**